keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
 
-![CreateBucket Workflow](images/create-bucket.png)
+The BucketAccessRequest is a request to create BucketAccess. This resource is designed to specify the Bucket for which the credentials provide access, and also for specifying access modes (rw, ro, wo) for that the application requires. More information about BucketAccessRequest is [here](#bucketaccessrequest)
 
-This workflow describes the automation supporting creating a new (greenfield) backend bucket. Accessing this bucket is covered in [Sharing COSI Created Buckets](#sharing-cosi-created-buckets). COSI determines that a request is for an existing bucket when `BucketRequest.bucketInstanceName` is specified.
+BucketSecret is the name of the secret that COSI will generate with credentials to access the bucket. The same secret name has to be set in the podSpec as a projected secret volume.
 
-Here is the workflow:
+```
+    BucketAccessRequest - bar1                                BucketAccessClass - bac1
+    |---------------------------------------|                 |----------------------------------|
+    | spec:                                 |                 | spec:                            |
+    |   bucketAccessClassName: bac1         |                 |   provisioner: s3.amazonaws.com  |
+    |   bucketRequestName: br1              |                 |   parameters:                    | 
+    |   credentialsSecretName: bucketcreds1 |                 |     key: value                   |
+    |---------------------------------------|                 |----------------------------------|
+```
 
-+ COSI central controller sees a new `BucketRequest` (BR).
-+ COSI gets the `BR.BucketClass` (directly or via the matching default) and verifies that the BR's namespace is allowed.
-+ If BR.bucketInstanceName is empty, COSI creates the associated `Bucket`, filling in fields from the `BucketClass` and BR. Else, we have a brownfield BR.
-+ The sidecar sees the new `Bucket` and gRPC calls the driver to create a backend bucket.
-+ COSI updates the BR and `Bucket` instance to reference each other.
+###### 2. COSI creates an intermediate BucketAccess object
 
-In addition, the cosi-node-adapter sees a new app pod references a BAR:
-+ cosi-node-adapter receives _NodeStageVolume_ request from kubelet.
-+ adapter add a finalizer to both the BAR.BA and the BAR.BR.
-+ the finalizer value is `pod.name`
+The ControllerManager creates a BucketAccess object by copying fields over from both BucketAccessRequest and BucketAccessClass. This step can only proceed if the BucketAccessClass and Bucket already exist. The Bucket's `BucketReady` condition should be true. This BucketAccess object is an intermediate object, with its condition `AccessGranted` set to false and credentialsSeret pointing to a `nil` value. This indicates that access credentials are yet to be generated in the OSP.
```

Comment:
What governs when this is allowed or forbidden?  What stops me from accessing your bucket?

Reply:
Right now, this is free for all. In the future, we are planning to add support for service account based authz/authn

Reply:
Within a single namespace, when is it appropriate to use 2 BARs, versus just sharing a secret?

Reply:
If the admin wants to be able to revoke access for one application without affecting another one, then two different BARs should be used.

Reply:
Might be worth a small note, eg.

"Within a namespace, on BAR+secret is generally sufficient, but cases which may want to control this more granularly can use multiple"